### PR TITLE
Spin up mainfork cluster

### PR DESCRIPTION
**Pull request by toliaqat**
We create two new nodes fork1 and fork2 that will be included in a mainfork cluster instance.

Mainfork is a cosmos snapshot of mainnet fork. The fork is created by[ cosmos-tinkerer ](https://github.com/agoric-labs/cosmos-genesis-tinkerer) repo. We don't deal about creation of the fork and tinkering of genesis in this repo. We only use the snapshots created by some other process and download them from gcloud storage for starting a cluster.

Currently, due to a known issue in tendermint, we cannot start a fork with single node. We use two nodes to create mainnet fork. Those two fork nodes are hard coded here as fork1 and fork2. These two nodes download their respective config from cloud storage to start agd.

### CODE REVIEW

File: bases/shared/entrypoint.sh [hunk 1]
```diff
@@ -400,6 +411,70 @@ get_ips() {
...
+    rm -rf $AGORIC_HOME/*
+
+    if [ ! -f "$MAINFORK_IMAGE_URL/$THIS_FORK-config-$MAINFORK_HEIGHT.tar.gz" ]; then
+        apt install -y axel
+        axel --quiet -n 10 -o "/state/$THIS_FORK-config-$MAINFORK_HEIGHT.tar.gz" "$MAINFORK_IMAGE_URL/$THIS_FORK-config-$MAINFORK_HEIGHT.tar.gz"
+        axel --quiet -n 10 -o "/state/agoric-$MAINFORK_HEIGHT.tar.gz" "$MAINFORK_IMAGE_URL/agoric-$MAINFORK_HEIGHT.tar.gz"
+
+        tar -xzf "/state/$THIS_FORK-config-$MAINFORK_HEIGHT.tar.gz" -C $AGORIC_HOME        
+        tar -xzf "/state/agoric-$MAINFORK_HEIGHT.tar.gz" -C $AGORIC_HOME
+    fi
```
Comment (michaelfig) [inline bases/shared/entrypoint.sh:469]: I recommend not using `hang` in this script except as controlled by the existence of `/state/hang`.  Maybe something like one of the following?

```suggestion
        # hang to debug if chain stops for any reason instead of restarting pods forever
        date > /state/hang
    fi
```

```suggestion
        # hang the first node to debug if it stops for any reason instead of restarting pods forever
        case "$THIS_FORK in
        *[^0-9]1) date > /state/hang ;;
        esac
    fi
```

File: bases/shared/entrypoint.sh [hunk 2]
```diff
@@ -739,6 +819,24 @@ case "$ROLE" in
...
+        hang
+        ;;
+    "fork2")
+        (WHALE_KEYNAME=whale POD_NAME=fork1 SEED_ENABLE=no NODE_ID='0663e8221928c923d516ea1e8972927f54da9edb' start_helper &)
+        fork_setup agoric2
+        export DEBUG="agoric,SwingSet:ls,SwingSet:vat"
+        start_chain --x-crisis-skip-assert-invariants --iavl-disable-fastnode false
+
+        # hang to debug if chain stops for any reason instead of restarting pods forever
+        hang
```
Comment (michaelfig) [inline bases/shared/entrypoint.sh:838]:
```suggestion
```

File: bases/shared/entrypoint.sh [hunk 3]
```diff
@@ -739,6 +819,24 @@ case "$ROLE" in
...
         start_chain --pruning everything
         ;;
+    "fork1")
+        (WHALE_KEYNAME=whale POD_NAME=fork1 SEED_ENABLE=no NODE_ID='0663e8221928c923d516ea1e8972927f54da9edb' start_helper &)
+        fork_setup agoric1
+        export DEBUG="agoric,SwingSet:ls,SwingSet:vat"
+        start_chain --x-crisis-skip-assert-invariants --iavl-disable-fastnode false
+        
+        # hang to debug if chain stops for any reason instead of restarting pods forever
+        hang
```
Comment (michaelfig) [inline bases/shared/entrypoint.sh:829]:
```suggestion
```

File: bases/shared/instagoric-server/server.js [hunk 4]
```diff
@@ -143,14 +146,22 @@ const getNetworkConfig = async () => {
   ap.peers[0] = ap.peers[0].replace(
     'validator-primary.instagoric.svc.cluster.local',
     svc.get('validator-primary-ext') ||
-      'validator-primary.instagoric.svc.cluster.local',
+      `${podname}.instagoric.svc.cluster.local`,
+  );
+  ap.peers[0] = ap.peers[0].replace(
+    'fb86a0993c694c981a28fa1ebd1fd692f345348b', `${NODE_ID}`,
```
Comment (michaelfig) [inline bases/shared/instagoric-server/server.js:152]: Why is this necessary and hardcoded?
```suggestion
    'fb86a0993c694c981a28fa1ebd1fd692f345348b', `${NODE_ID}`,
```

**Review by michaelfig — APPROVED** [review]
Looking good.  Please address or rebut comments, then feel free to merge.

File: bases/shared/instagoric-server/server.js (hunk 4 above)
Comment (michaelfig) [reply bases/shared/instagoric-server/server.js:152]: Okay, so it's a hardcoded `network_config.json` used in instagoric.  Thanks for explaining it.

File: bases/shared/entrypoint.sh (hunk 3 above)
Comment (toliaqat) [reply bases/shared/entrypoint.sh]: I have removed hang in these places.

File: bases/shared/instagoric-server/server.js (hunk 4 above)
Comment (toliaqat) [reply bases/shared/instagoric-server/server.js:152]: Thank you for the review.
